openmc/deplete/flux_operator.py

```diff
@@ -0,0 +1,601 @@
+from .chain import REACTIONS
+from .openmc_operator import OpenMCOperator, _distribute
+from .results import Results
+from .helpers import ChainFissionHelper, ConstantFissionYieldHelper, SourceRateHelper
+
+_valid_rxns = list(REACTIONS)
+_valid_rxns.append('fission')
+
+def generate_1g_cross_sections(model,
+                               reaction_domain,
+                               reactions=['(n,gamma)',
+                                          '(n,2n)',
+                                          '(n,p)',
+                                          '(n,a)',
+                                          '(n,3n)',
+                                          '(n,4n)',
+                                          'fission'],
+                               energy_bounds=(0, 20e6),
+                               write_to_csv=False,
+                               filename='micro_xs.csv'):
+    """Helper function to generate a one-group cross-section dataframe  using
+    OpenMC. Note that the ``openmc`` executable must be compiled.
+
+    Parameters
+    ----------
+    model : openmc.model.Model
+        OpenMC model object. Must contain geometry, materials, and settings.
+    reaction_domain : openmc.Material or openmc.Cell or openmc.Universe or openmc.RegularMesh
+        Domain in which to tally reaction rates.
+    reactions : list of str, optional
+        Reaction names to tally
+    energy_bound : 2-tuple of float, optional
+        Bounds for the energy group.
+    write_to_csv : bool, optional
+        Option to write the DataFrame to a `.csv` file.
+    filename : str
+        Name for csv file. Only applicable if ``write_to_csv == True``
+
+    Returns
+    -------
+    None or pandas.DataFrame
+
+    """
+    groups = EnergyGroups(energy_bounds)
+
+    # Set up the reaction tallies
+    original_tallies = model.tallies
+    tallies = openmc.Tallies()
+    xs = {}
+    for rxn in reactions:
+        if rxn == 'fission':
+            xs[rxn] = FissionXS(domain=reaction_domain, groups=groups, by_nuclide=True)
+        else:
+            xs[rxn] = ArbitraryXS(rxn, domain=reaction_domain, groups=groups, by_nuclide=True)
+        tallies += xs[rxn].tallies.values()
+
+    model.tallies = tallies
+    statepoint_path = model.run()
+
+    # Revert to the original tallies
+    model.tallies = old_tallies
+
+    with openmc.StatePoint(statepoint_path) as sp:
+        for rxn in xs:
+            xs[rxn].load_from_statepoint(sp)
+
+    # Build the DataFrame
+    micro_xs = pd.DataFrame()
+    for rxn in xs:
+        df = xs[rxn].get_pandas_dataframe(xs_type='micro')
+        df.index = df['nuclide']
+        df.drop(['nuclide', xs[rxn].domain_type, 'group in', 'std. dev.'], axis=1, inplace=True)
+        df.rename({'mean':rxn}, axis=1, inplace=True)
+        micro_xs = pd.concat([micro_xs, df], axis=1)
+
+    if write_to_csv:
+        micro_xs.to_csv(filename)
+
+    return micro_xs
+
+class FluxDepletionOperator(OpenMCOperator):
```

Comment:
I'm reconsidering whether this name is apt. What do you think of `IndependentOperator`? I think the "Depletion" part is probably not necessary since it's already namespaced in `openmc.deplete`.

Reply:
Another alternative: `ExternalOperator` (i.e., transport is run externally)

Reply:
I agree. It's fine as a preliminary name but we should pick something more polished before this gets merged.

Reply:
I think `ExternalOperator` makes the most sense